plugins/interactive/displayserver.go

```diff
@@ -0,0 +1,177 @@
+package interactive
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"sync"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/gorilla/websocket"
+	"github.com/taskcluster/taskcluster-worker/engines"
+	"github.com/taskcluster/taskcluster-worker/plugins/interactive/displayconsts"
+)
+
+type displayServer struct {
+	m        sync.Mutex
+	sandbox  engines.Sandbox
+	log      *logrus.Entry
+	done     chan struct{}
+	handlers []*DisplayHandler
+}
+
+func newDisplayServer(sandbox engines.Sandbox, log *logrus.Entry) *displayServer {
+	return &displayServer{
+		log:     log,
+		sandbox: sandbox,
+		done:    make(chan struct{}),
+	}
+}
+
+func (s *displayServer) Abort() {
+	s.m.Lock()
+	defer s.m.Unlock()
+
+	select {
+	case <-s.done:
+	default:
+		close(s.done)
+	}
```

Comment:
We can't call close() on a channel twice, so we check that it has not already been called.
This works because I hold a mutex whenever I do this...